packages/block-library/src/search/edit.js

```diff
@@ -208,11 +210,14 @@ export default function SearchEdit( {
 		// If the input is inside the wrapper, the wrapper gets the border color styles/classes, not the input control.
 		const textFieldClasses = classnames(
 			'wp-block-search__input',
-			isButtonPositionInside ? undefined : borderProps.className
+			isButtonPositionInside ? undefined : borderProps.className,
+			typographyProps.className
 		);
-		const textFieldStyles = isButtonPositionInside
-			? { borderRadius }
-			: borderProps.style;
+		const textFieldStyles = {
+			...( isButtonPositionInside
+				? { borderRadius }
+				: borderProps.style ),
+		};
```

Comment:
After rolling back previous changes, I don't think we need this change anymore.

Reply:
Thanks for testing again @aaronrobertshaw, and for spotting the unnecessary code snippet here.

 It made me see something else: I'd forgotten to roll back the application of the font size and family preset classnames to the individual elements. I've applied them to the wrapper only now.

You shouldn't see any differences in testing.

If it looks okay on Monday, please feel free to merge this PR. 🙇 🙇 🙇 🙇